app/views/play/level/tome/SpellView.coffee

```diff
@@ -796,24 +805,20 @@ module.exports = class SpellView extends CocoView
       else
         finishUpdatingAether(aether)
 
-  # NOTE! Because this alone causes the doctype annotation to flicker,
-  # all info annotations have been hidden with CSS in spell.sass
-  # If we ever want info annotations back, we need to remove that.
-  #
-  # This function itself removes the unwanted annotations on a later tick.
-  onChangeAnnotation: (event, session) ->
-    unfilteredAnnotations = session.getAnnotations()
-    filteredAnnotations = _.reject unfilteredAnnotations, (annotation) ->
-      annotation.text is 'Start tag seen without seeing a doctype first. Expected e.g. <!DOCTYPE html>.'
-    if filteredAnnotations.length < unfilteredAnnotations.length
-      session.setAnnotations(filteredAnnotations)
+  clearWebDevErrors: ->
+    nonWebDevAnnotations = _.reject @aceSession.getAnnotations(), (annotation) -> annotation.createdBy is 'web-dev-iframe'
+    @reallySetAnnotations nonWebDevAnnotations
+
+    problemsToClear = _.filter @problems, (p) -> p.createdBy is 'web-dev-iframe'
+    problemsToClear.forEach (problem) -> problem.destroy()
+    @problems = _.reject @problems, (p) -> p.createdBy is 'web-dev-iframe'
```

Comment:
You know a good way to use it given that one's a `.filter` and one's a `.reject`? Doing the destruction inside the filter callback would work but feels weird(ly non-functional)